v0/destinations/google_adwords_remarketing_lists/transform.js

```diff
@@ -132,66 +135,61 @@ const populateIdentifiers = (attributeArray, { Config }) => {
+    if (properties.includes(key)) {
+      const userIdentifiersList = populateIdentifiers(
+        listData[key],
+        destination
+      );
+      const outputPayload = constructPayload(message, offlineDataJobsMapping);
+      outputPayload.operations = [];
+      // breaking the userIdentiFier array in chunks of 20
+      const userIdentifierChunks = returnArrayOfSubarrays(
+        userIdentifiersList,
+        20
+      );
+      // putting each chunk in different create/remove operations
+      switch (key) {
+        case "add":
+          // for add operation
+          userIdentifierChunks.forEach(element => {
+            const operations = {
+              create: {}
+            };
+            operations.create.userIdentifiers = element;
+            outputPayload.operations.push(operations);
+          });
+          outputPayloads = { ...outputPayloads, create: outputPayload };
+          break;
+        case "remove":
+          // for remove operation
+          userIdentifierChunks.forEach(element => {
+            const operations = {
+              remove: {}
+            };
+            operations.remove.userIdentifiers = element;
+            outputPayload.operations.push(operations);
+          });
+          outputPayloads = { ...outputPayloads, remove: outputPayload };
+          break;
+        default:
+      }
+    } else {
+      logger.info(
+        `listData "${key}" is not valid. Supported types are "add" and "remove"`
+      );
+    }
   });
 
-  return outputPayload;
-  /**
-   * This portion is to support remove and create both.
-   * If we need to support for remove too then need to uncomment this section
-   * and update addUserToJob function in util to handle add and remove both.
-   * Now we are directly sending the the body.JSON but to support remove we have to make two calls
-   * by setting data = body.JSON[0] and data = body.JSON[1] synultaneously.
-   */
-  // const outputPayloads = [];
-  // const typeOfOperation = Object.keys(listData);
-  // typeOfOperation.forEach(key => {
-  //   const userIdentidtifiersList = populateIdentifiers(
-  //     listData[`${key}`],
-  //     destination
-  //   );
-  //   const outputPayload = constructPayload(message, offlineDataJobsMapping);
-  //   outputPayload.operations = [];
-  //   // breaking the userIdentiFier array in chunks of 20
-  //   const userIdentifierChunks = returnArrayOfSubarrays(
-  //     userIdentidtifiersList,
-  //     20
-  //   );
-  //   // putting each chunk in different create/remove operations
-  //   userIdentifierChunks.forEach(element => {
-  //     const operations = {
-  //       [`${key}`]: {}
-  //     };
-  //     operations[`${key}`].userIdentifiers = element;
-  //     outputPayload.operations.push(operations);
-  //   });
-  //   outputPayloads.push(outputPayload);
-  // });
-  // return outputPayloads;
+  return outputPayloads;
 };
 
 const processEvent = async (metadata, message, destination) => {
```

Comment:
Do we have test cases to cover all the validations performed here?

Reply:
have added all the test cases.

Reply:
Can you please point me to each of those TCs?

Reply:
Do not have the exact list rn but can be found in test cases input/output files. Mainly for deletion support, different scenarios of how we pass data have been added and few mentioned below are:
* breaking the user identifier array in chunks of 20
* If neither add nor remove keyword is provided then it throws a custom error.
* only passing remove operation etc.
For the `add` operation @ItsSudip has already added all the custom errors which are being handled and its scenarios.